client: print full query in case of failures (test hints mismatch or server exception)

azat opened the pull request:
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

azat committed:
client: print full query in case of test hints failures

azat committed:
client: print full query in case of server exception

azat committed:
client: print query on client exception only in non-interactive mode

kssenii commented on programs/client/Client.cpp:
```diff
@@ -1031,19 +1031,30 @@ class Client : public Poco::Util::Application
+            }
+            else
+            {
+                fmt::print(stderr, "(query: {})\n", full_query);
```
May be it will be better to print full query under an option? because what if it is some huge multiquery?

azat replied:
The query was printed for the client error case anyway, plus I'm not sure that adding an option just for this worth it.

NOTE: that it will make CI logs more informative, since right now they do not report the query in case of server exception or test hint mismatch.

kssenii replied:
Ok, agree.

azat committed:
tests: update reference after client started printing query

azat committed:
tests: update more references after client started printing query

azat committed:
tests/queries/0_stateless/01175_distributed_ddl_output_mode_long: use unique tmp file

azat committed:
tests: update 01305_replica_create_drop_zookeeper after client query printing

azat committed:
tests: mark 00763_lock_buffer as long

https://clickhouse-test-reports.s3.yandex.net/28827/58266e4ed6db77109e6237ecd2612d34c0b19516/functional_stateless_tests_flaky_check_(address).html#fail1

azat commented:
>Functional stateless tests (thread) — fail: 1, passed: 3328, skipped: 27

```
2021-09-11 16:59:57 00652_replicated_mutations_zookeeper:                                   [ FAIL ] 19.03 sec. - having stderror:
2021-09-11 16:59:57 [b7e855acf283] 2021.09.11 16:59:42.265576 [ 29389 ] {517c32f3-f4c3-43d0-9156-f7ade4319b64} <Error> executeQuery: Code: 341. DB::Exception: Mutation 0000000000 was killed. (UNFINISHED) (version 21.11.1.8053) (from [::1]:59958) (comment: '/usr/share/clickhouse-test/queries/0_stateless/00652_replicated_mutations_zookeeper.sh') (in query: ALTER TABLE mutations_r1 DELETE WHERE x = 1 SETTINGS mutations_sync = 2), Stack trace (when copying this message, always include the lines below):
```

#28889

tavplubix commented:
It breaks test `01150_ddl_guard_rwr`:
https://clickhouse-test-reports.s3.yandex.net/0/1d05fab295d425dbdb549b2b5f80801f1d59f4bc/functional_stateless_tests_(release,_databaseordinary).html#fail1
